test/collectionUtils/AnyResourceTransformer.test.ts

```diff
@@ -0,0 +1,93 @@
+import AnyResourceTypeTransformer from "../../src/collectionUtils/AnyResourceTypeTransformer";
+import EntityDefinitionBuilder from "../model/dataTypes/EntityDefinitionBuilder";
+import EntityDefinition from "../../src/model/dataTypes/EntityDefinition";
+import MemberVariable from "../../src/model/dataTypes/MemberVariable";
+import DataType from "../../src/model/dataTypes/DataType";
+import EntityImports from "../../src/model/dataTypes/EntityImports";
+import FilePath from "../../src/model/dataTypes/FilePath";
+
+describe("AnyResourceTypeTransformer", () => {
+  const baseDir = FilePath.getInstance("/tmp");
+  const transformer = new AnyResourceTypeTransformer(baseDir);
+  let regularEntity: EntityDefinition;
+  let resourceMemberEntity: EntityDefinition;
+  let resourceChoiceEntity: EntityDefinition;
+
+  beforeEach(() => {
+    const builder = new EntityDefinitionBuilder();
+    regularEntity = builder.buildEntityDefinition();
+
+    const iResourceType = DataType.getInstance("FHIR", "IResource", baseDir);
+    const otherType = DataType.getInstance("FHIR", "SomethingElse", baseDir);
+
+    const resourceMember = new MemberVariable(
+      iResourceType,
+      "resourceMember",
+      false,
+      undefined,
+      false,
+      [otherType]
+    );
+    builder.imports = new EntityImports([iResourceType]);
+    builder.memberVariables = [resourceMember];
+    resourceMemberEntity = builder.buildEntityDefinition();
+
+    const choiceMember = new MemberVariable(
+      otherType,
+      "choiceMember",
+      false,
+      undefined,
+      false,
+      [iResourceType, otherType]
+    );
+    builder.memberVariables = [choiceMember];
+    resourceChoiceEntity = builder.buildEntityDefinition();
+  });
+
+  it("should return the same instance if no members have IResource members", () => {
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5923420)